pkg/graveler/graveler.go

```diff
@@ -541,6 +541,11 @@ type KeyValueStore interface {
 	// Set stores value on repository / branch by key. nil value is a valid value for tombstone
 	Set(ctx context.Context, repository *RepositoryRecord, branchID BranchID, key Key, value Value, opts ...SetOptionsFunc) error
 
+	// UpdateObjectMetadata atomically runs update on repository /
+	// branch by key.  (Of course, if entry is only on committed, the
+	// updated entry will still be created (atomically) on staging.)
+	UpdateObjectMetadata(ctx context.Context, repository *RepositoryRecord, branchID BranchID, key Key, update ValueUpdateFunc, opts ...SetOptionsFunc) error
```

Comment:
I don't know if you meant to write it this way, but you can update more than just the metadata. If it's intended, let's rename the func. If not, let's rewrite the logic.
```suggestion
	UpdateObject(ctx context.Context, repository *RepositoryRecord, branchID BranchID, key Key, update ValueUpdateFunc, opts ...SetOptionsFunc) error
```